File: fluentcore/src/main/java/com/mcreater/fxfluent/util/NumberUtil.java
package com.mcreater.fxfluent.util;

public class NumberUtil {
    public static double lim(double res, double up, double down) {
        double swap;
        if (down > up) {
            swap = up;
            up = down;
            down = swap;
        }

        return Math.max(down, Math.min(up, res));
    }

    public static int lim(int res, int up, int down) {
        int swap;
        if (down > up) {
            swap = up;
            up = down;
            down = swap;
        }

        return Math.max(down, Math.min(up, res));
    }
}

File: fluentcore/src/main/java/com/mcreater/fxfluent/util/NumberUtil.kt
package com.mcreater.fxfluent.util

import kotlin.math.max
import kotlin.math.min

class NumberUtil {
    companion object {
        @JvmStatic
        fun lim(res: Double, up: Double, down: Double): Double {
            var upit = up
            var downit = down
            if (downit > upit) {
                upit = downit.apply {downit = upit}
            }
            return max(downit, min(upit, res))
        }

        @JvmStatic
        fun lim(res: Int, up: Int, down: Int): Int {
            var upit = up
            var downit = down
            if (downit > upit) {
                upit = downit.apply {downit = upit}
            }
            return max(downit.toDouble(), min(upit.toDouble(), res.toDouble())).toInt()
        }
    }
}
